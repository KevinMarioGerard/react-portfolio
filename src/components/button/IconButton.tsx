import * as React from 'react';

type IconButtonProps = {
    /**
     * Icon to render
     */
    icon: string;

    /**
     * Redirection url on button click
     */
    redirectionUrl: string;
};

const IconButton: React.FC<IconButtonProps> = ({ icon, redirectionUrl, ...rest }) => (
    <div
        className="icon icon__container"
        onClick={React.useCallback(() => {
            window.open(redirectionUrl);
        }, [redirectionUrl])}
        {...rest}>
        <i className={`fa ${icon}`} />
    </div>
);

export default IconButton;
