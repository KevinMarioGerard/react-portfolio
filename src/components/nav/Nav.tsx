import * as React from 'react';
import ProfilePic, { ProfilePicProps } from '../profilePic/ProfilePic';

type NavHeaderProps = {
    /**
     * Nav Header label to render
     */
    header: string;

    /**
     * Nav Header caption to render
     */
    caption?: string;

    /**
     * Nav footer icon buttons to render
     */
    footerIcons?: React.ReactNode;
} & ProfilePicProps;

const NavHeader: React.FC<NavHeaderProps> = ({
    header = '',
    caption,
    profilePicSrc,
    footerIcons,
}) => (
    <div className="nav__header">
        <ProfilePic profilePicSrc={profilePicSrc} />
        <header>{header}</header>
        {caption ? (
            <div className="nav__header__caption">
                <span>{caption}</span>
            </div>
        ) : null}
        {footerIcons ? <div className="nav__header__footer">{footerIcons}</div> : null}
    </div>
);

NavHeader.displayName = 'Nav.Header';

const NavItemsList: React.FC = props => <div className="nav__items-list">{props.children}</div>;

NavItemsList.displayName = 'Nav.ItemsList';

type NavItemProps = {
    /**
     * Nav Item Label
     */
    label: string;

    /**
     * Nav Item id
     */
    id: string;

    /**
     * Nav Item icon
     */
    icon?: string;
};

const NavItem: React.FC<NavItemProps> = ({ label = '', id = '', icon = '', ...rest }) => (
    <div className="nav__item" {...rest}>
        <i className={`nav__item__icon ${icon}`} />
        <a href={`#${id}`}>{label}</a>
    </div>
);

NavItem.displayName = 'Nav.Item';

type Statics = {
    Header: typeof NavHeader;
    Item: typeof NavItem;
    ItemsList: typeof NavItemsList;
};

const Nav: React.FC & Statics = props => <nav className="shell__nav">{props.children}</nav>;

Nav.displayName = 'Nav';

Nav.Header = NavHeader;
Nav.ItemsList = NavItemsList;
Nav.Item = NavItem;

export default Nav;
