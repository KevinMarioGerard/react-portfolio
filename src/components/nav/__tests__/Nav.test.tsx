import { render } from '@testing-library/react';
import * as React from 'react';
import Nav from '../Nav';
import IconButton from '../../button/IconButton';

describe('Nav Header Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<Nav.Header header="Test Header" profilePicSrc="test-image" />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(
            <Nav.Header header="Test Header" caption="Test Caption" profilePicSrc="test-image" />
        );

        expect(container).toMatchSnapshot();
    });

    it('should render footer icons', () => {
        const { container } = render(
            <Nav.Header
                header="Test Header"
                caption="Test Caption"
                profilePicSrc="test-image"
                footerIcons={
                    <>
                        <IconButton icon="fa-github" redirectionUrl="test-github-url" />
                        <IconButton icon="fa-linkedin" redirectionUrl="test-linkedin-url" />
                    </>
                }
            />
        );

        expect(container).toMatchSnapshot();
    });
});

describe('Nav Items List Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(
                <Nav.ItemsList>
                    <Nav.Item label="Test" id="test" />
                </Nav.ItemsList>
            );
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(
            <Nav.ItemsList>
                <Nav.Item label="Test" id="test" />
            </Nav.ItemsList>
        );

        expect(container).toMatchSnapshot();
    });
});

describe('Nav Item Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<Nav.Item label="Test" id="test" />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(<Nav.Item label="Test" id="test" />);

        expect(container).toMatchSnapshot();
    });
});

describe('Nav Footer Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<Nav.Footer />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(<Nav.Footer>Test Navigation Footer</Nav.Footer>);

        expect(container).toMatchSnapshot();
    });
});
