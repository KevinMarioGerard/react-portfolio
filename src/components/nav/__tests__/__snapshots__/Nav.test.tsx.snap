// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Footer Component tests should match snapshot 1`] = `
<div>
  <footer
    class="nav__footer"
  >
    Test Navigation Footer
  </footer>
</div>
`;

exports[`Nav Header Component tests should match snapshot 1`] = `
<div>
  <div
    class="nav__header"
  >
    <div
      class="profile__container"
    >
      <img
        alt="Profile Pic"
        class="profile__pic"
        src="test-image"
      />
    </div>
    <header>
      Test Header
    </header>
    <div
      class="nav__header__caption"
    >
      <span>
        Test Caption
      </span>
    </div>
  </div>
</div>
`;

exports[`Nav Header Component tests should render footer icons 1`] = `
<div>
  <div
    class="nav__header"
  >
    <div
      class="profile__container"
    >
      <img
        alt="Profile Pic"
        class="profile__pic"
        src="test-image"
      />
    </div>
    <header>
      Test Header
    </header>
    <div
      class="nav__header__caption"
    >
      <span>
        Test Caption
      </span>
    </div>
    <div
      class="nav__header__footer"
    >
      <div
        class="icon icon__container"
      >
        <i
          class="fa fa-github"
        />
      </div>
      <div
        class="icon icon__container"
      >
        <i
          class="fa fa-linkedin"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Nav Item Component tests should match snapshot 1`] = `
<div>
  <div
    class="nav__item"
  >
    <i
      class="nav__item__icon "
    />
    <label>
      Test
    </label>
  </div>
</div>
`;

exports[`Nav Items List Component tests should match snapshot 1`] = `
<div>
  <div
    class="nav__items-list"
  >
    <div
      class="nav__item"
    >
      <i
        class="nav__item__icon "
      />
      <label>
        Test
      </label>
    </div>
  </div>
</div>
`;
